pkg/rekor/identity.go

```diff
@@ -335,3 +340,37 @@ func oidMatchesPolicy(cert *x509.Certificate, oid asn1.ObjectIdentifier, extensi
 
 	return false, nil, "", nil
 }
+
+// writeIdentitiesBetweenCheckpoints monitors for given identities between two checkpoints and writes any found identities to file.
+func writeIdentitiesBetweenCheckpoints(logInfo *models.LogInfo, prevCheckpoint *util.SignedCheckpoint, checkpoint *util.SignedCheckpoint, monitoredValues identity.MonitoredValues, rekorClient *client.Rekor, outputIdentitiesFile string) error {
+	// Get log size of inactive shards
+	totalSize := 0
+	for _, s := range logInfo.InactiveShards {
+		totalSize += int(*s.TreeSize)
+	}
+	startIndex := int(prevCheckpoint.Size) + totalSize - 1 //nolint: gosec // G115, log will never be large enough to overflow
+	endIndex := int(checkpoint.Size) + totalSize - 1       //nolint: gosec // G115
+
+	// Search for identities in the log range
+	if len(monitoredValues.CertificateIdentities) > 0 || len(monitoredValues.Fingerprints) > 0 || len(monitoredValues.Subjects) > 0 || len(monitoredValues.OIDMatchers) > 0 {
```

Comment:
Same as https://github.com/sigstore/rekor-monitor/pull/483#discussion_r1801207248, let's extract this to a function and reuse.